src/rmview/screenstream/screenshare.py

```diff
@@ -11,15 +11,62 @@
 import sys
 import os
 import logging
-
-from twisted.internet.protocol import Protocol
+import configparser
+import jwt
+import io
+import socket
+import hashlib
+from twisted.internet.protocol import Protocol,DatagramProtocol
 from twisted.internet import protocol, reactor, ssl
 from twisted.application import internet, service
 
+
 from .common import *
 
 log = logging.getLogger('rmview')
 
+# the screenshare vnc auth uses udp broadcasts
+class ChallengeReaderProtocol(DatagramProtocol):
+  clients = {}
+  started = False
+
+  def __init__(self, screenShareStream):
+    self.screenShareStream = screenShareStream
+
+  def datagramReceived(self, datagram, host):
+    reader = io.BytesIO(datagram)
+
+    # the timestamp is needed for the challenge
+    timestamp = reader.read(8)
+    tounx, = unpack("!Q", timestamp)
+    if timestamp in self.clients:
+      log.debug(f"skipping challenge {tounx}")
+      return
+    log.info(f"received timestamp challenge {tounx}")
+
+    #TODO: just quick and dirty way to run the vnc after the challenge is received
+    if not self.started:
+      self.started = True
+      self.screenShareStream.runVnc(timestamp)
```

Comment:
Any reason why we should not stop listening (with `self.transport.stopListening()`) after calling `runVnc`?

Reply:
same as above, after restarting screensharing a new challenge is generated

Reply:
it seems it works with the old challenge, but I didn't know how to stop the udp listener

Reply:
So I guess a first attempt at simplifying this is to do `self.transport.stopListening()` after `runVnc` and maybe remove the parsing of the ip:port list (as it seems unused?).
Also, the `started` flag could maybe be maintained in the client and maintained by `runVnc`? Maybe we could even use `self.factory` as a proxy for it (if there is a factory we already started the client).

Reply:
one could replace the whole block here with something like `if not self.callback(timestamp): self.transport.stopListening()` where self.callback saves a `callback` parameter one passes to the constructor instead of `screenShareStream`